enclave/core/sgx/calls.c

```diff
@@ -171,6 +172,10 @@ static oe_result_t _handle_init_enclave(uint64_t arg_in)
             /* Initialize the CPUID table before calling global constructors. */
             OE_CHECK(oe_initialize_cpuid());
 
+            /* Initialize the xstate settings
+             * Depends on TD and sgx_create_report, so can't happen earlier */
+            OE_CHECK(oe_set_is_xsave_supported());
```

Comment:
Can the host cause an exception in the enclave before xsave_supported is set?
Does our exception handling infrastructure need to do anything in regards to xsave supported?

Reply:
The working assumption is the OS can always cause an enclave to fault at any time. However, AEX handles the XSAVE state area via the XFRM settings of the enclave and also provides synthetic state for extended features, which is unrelated to the flags that our runtime sets, so it is not my understanding that this needs to be mirrored in the exception handling code path. 

There's a different issue in the nested callback to the exception dispatcher where we are not preserving and restoring the xsave state (`oe_snap_current_context`/`oe_continue_execution`),but that's a separate work item to fix preservation of the XSAVE state everywhere (in host on ecall, in enclave on ocall, in exception dispatching).